src/bin/ipc_latency.rs

```diff
@@ -43,30 +44,75 @@ impl portus::serialize::AsRawMsg for TimeMsg {
         Ok(TimeMsg(time::Timespec::new(sec, nsec)))
     }
 }
+#[derive(Debug)]
+struct DetTimeMsg(time::Timespec, time::Timespec);
+impl portus::serialize::AsRawMsg for DetTimeMsg {
+    fn get_hdr(&self) -> (u8, u32, u32) {
+        (0xff1, portus::serialize::HDR_LENGTH + 16 + 8, 0) 
+    }
+
+    fn get_u32s<W: Write>(&self, _: &mut W) -> portus::Result<()> {
+        Ok(())
+    }
+
+    fn get_u64s<W: Write>(&self, _: &mut W) -> portus::Result<()> {
+        Ok(())
+    }
+
+    fn get_bytes<W: Write>(&self, w: &mut W) -> portus::Result<()> {
+        let mut msg = [0u8; 24]; // one i64 and one i32
+        LittleEndian::write_i64(&mut msg[0..8], self.0.sec);
+        LittleEndian::write_i32(&mut msg[8..12], self.0.nsec);
+        LittleEndian::write_i64(&mut msg[12..20], self.1.sec);
+        LittleEndian::write_i32(&mut msg[20..24], self.1.nsec);
+        w.write_all(&msg[..])?;
+        Ok(())
+    }
+
+    fn from_raw_msg(msg: portus::serialize::RawMsg) -> portus::Result<Self> {
+        let b = msg.get_bytes()?;
+        let up_sec = LittleEndian::read_i64(&b[0..8]);
+        let up_nsec = LittleEndian::read_i32(&b[8..12]);
+        let down_sec = LittleEndian::read_i64(&b[12..20]);
+        let down_nsec = LittleEndian::read_i32(&b[20..24]);
+        Ok(DetTimeMsg(time::Timespec::new(up_sec, up_nsec), time::Timespec::new(down_sec, down_nsec)))
+    }
+}
 
-use portus::serialize::AsRawMsg;
 use std::sync::mpsc;
-fn bench<T: Ipc>(b: &BackendSender<T>, mut l: Backend<T>, iter: u32) -> Vec<Duration> {
-    (0..iter)
+use portus::serialize::AsRawMsg;
+fn bench<T: Ipc>(
+    b: BackendSender<T>,
+    mut l: Backend<T>,
+    iter: u32,
+) -> Vec<Duration> {
+
+    let res = (0..iter)
         .map(|_| {
+
             let then = time::get_time();
             let msg = portus::serialize::serialize(&TimeMsg(then)).expect("serialize");
             b.send_msg(&msg[..]).expect("send ts");
-
-            if let portus::serialize::Msg::Other(raw) = l.next().expect("receive echo") {
+            if let portus::serialize::Msg::Other(raw) =
+                l.next().expect("receive echo")
+            {
                 let then = TimeMsg::from_raw_msg(raw).expect("get time from raw");
                 time::get_time() - then.0
             } else {
                 panic!("wrong type");
             }
         })
-        .collect()
+        .collect();
+    return res;
+
 }
 
+
+
 macro_rules! netlink_bench {
-    ($name: ident, $mode: ident) => {
+    ($name: ident, $mode: ident) => (
         #[cfg(target_os = "linux")] // netlink is linux-only
-        fn $name(iter: u32) -> Vec<Duration> {
+        fn $name(iter: u32) -> Vec<(Duration, Duration, Duration)> {
```

Comment:
might be worth it to define a type for `(Duration, Duration, Duration)`

Reply:
Yea I think that's a good idea so it's clear what each field represents.

Reply:
I may define a struct for the purpose I mentioned above.

Reply:
not strictly necessary, but might be better to name the fields just so it's clear, just wrapping it in a type doesn't really add any information about what the individual fields mean